From 86ea7bdbaac34505b29785efc1d233659babd659 Mon Sep 17 00:00:00 2001
From: Thierry Fournier <tfournier@arpalert.org>
Date: Fri, 19 Feb 2016 18:30:13 +0100
Subject: [PATCH 49/67] BUG/MINOR: lua: Useless copy

A value is copied two time in teh stack, but only one is usefull. The
second copy leaves unused in the stack and take some room for noting.

This path removes the second copy.

Must be backported in 1.6
(cherry picked from commit 9d5fb6d6a01331ac3fca3ff16b832697abe6e335)
---
 src/hlua.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/hlua.c b/src/hlua.c
index 991edf1..e89c83f 100644
--- a/src/hlua.c
+++ b/src/hlua.c
@@ -6963,7 +6963,6 @@ void hlua_init(void)
 
 	/* Register previous table in the registry with reference and named entry. */
 	lua_pushvalue(gL.T, -1); /* Copy the -1 entry and push it on the stack. */
-	lua_pushvalue(gL.T, -1); /* Copy the -1 entry and push it on the stack. */
 	lua_setfield(gL.T, LUA_REGISTRYINDEX, CLASS_SOCKET); /* register class socket. */
 	class_socket_ref = luaL_ref(gL.T, LUA_REGISTRYINDEX); /* reference class socket. */
 
-- 
2.4.10

